rosetta-source/src/main/rosetta/product-template-enum.rosetta

```diff
@@ -13,6 +13,11 @@ enum OptionTypeEnum: <"The enumerated values to specify the type of the option.
     Receiver <"A 'receiver' option: If you buy a 'receiver' option you have the right but not the obligation to enter into the underlying swap transaction as the 'fixed' rate/price receiver and pay float.">
     Straddle <"A straddle strategy, which involves the simultaneous buying of a put and a call of the same underlier, at the same strike and same expiration date">
 
+enum OptionExerciseStyleEnum:
+    European <"Single Exercise">
+    Bermuda <"Multiple specified exercise dates">
+    American <"Continuous exercise over a range of dates">
```

Comment:
Suggestion: Rename enum values to `Single`, `Multiple` and `Continuous` (market-agnostic) and add a reference to the "American", "European" and "Bermuda" terminology in the description of each value instead.

To be discussed with the wider group.

Reply:
Okay, will keep this in mind. For now, I will keep it as is until a consensus is reached.

Reply:
Based on DPBE discussion, this change doesn't have consensus. On this basis I would advise to carry on with your PR as-is and leave that question to a future change.


I do believe the change has merits and is relevant in the CDM *today* (not just for ture use-cases), because the harmonised `ExerciseTerms` object is used in context other than `OptionPayout`: e.g. `ExtendibleProvision` and `CancelableProvision`.

I would leave it to @tomhealey-icma and @chrisisla to make the case in future.

Reply:
My suggestion - which I think concurs with the findings of the DPBE - is that we should leave the enum as _European_, _American_ and _Bermuda_. Although making the terms agnostic is a nice idea I think in this case the exercise types are standard and recognised throughout the industry.

Reply:
@chrisisla I've had some further thoughts about this..

In short, I think for the "non-option" use case (e.g. cancellable or extendible provisions), there simply _isn't_ any need for a enum value because the rest of the object says what it is:
* if there is only an expiration date and no commendemnt date it's a single exercise
* if there are exercise dates it's multiple etc.

So indeed let's not invent a new set of enum values - as you said these exercise types are industry standard for options. Instead we make the `style` attribute optional, and only require it as part of the `OptionPayout`. So for the extendible / cancellable provision case, you can omit the `style` attribute whose values are meaningless in that context.

We still need to enforce some self-consistency constraints when the `style` attribute exists. And even when absent, we need to ensure the correct mutual exclusion conditions between the `commencementDate`, `expirationDate` and `exerciseDates` attributes. + @rogerguitart

Reply:
@chrisisla @lolabeis 
I will keep the current `style` enum naming of {American, European, Bermuda} given the feeback we've received. 

Regarding your points Leo, I have made `style` optional so that for `cancelableProvision` and `extendibleProvision` it can be omitted, and have added an `OptionStylePresent` condition in `OptionPayout` to make sure `style` is present in this case.

With respect to the constraints to the enumeration, there already exist the conditions in `ExerciseTerms` which control for the expected fields depending on whether Bermuda, European, or American has been selected:

![image](https://github.com/finos/common-domain-model/assets/141412602/f01675d3-4743-483a-a389-481bf9e1fa7c)

Please let me know if these are sufficient or if you believe there should be any additional ammendments to be made.

Note this PR has been closed due to conflicts - a new one has been generated with the relevant ammendments: #2687